darts/models/forecasting/nhits.py

```diff
@@ -0,0 +1,736 @@
+            object. Check the `PL Trainer documentation
+            <https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html>`_ for more information about the
+            supported kwargs. Default: ``None``.
+            With parameter ``"callbacks"`` you can add custom or PyTorch-Lightning built-in callbacks to Darts'
+            :class:`TorchForecastingModel`. Below is an example for adding EarlyStopping to the training process.
+            The model will stop training early if the validation loss `val_loss` does not improve beyond
+            specifications. For more information on callbacks, visit:
+            `PyTorch Lightning Callbacks
+            <https://pytorch-lightning.readthedocs.io/en/stable/extensions/callbacks.html>`_
+
+            .. highlight:: python
+            .. code-block:: python
+
+                from pytorch_lightning.callbacks.early_stopping import EarlyStopping
+
+                # stop training when validation loss does not decrease more than 0.05 (`min_delta`) over
+                # a period of 5 epochs (`patience`)
+                my_stopper = EarlyStopping(
+                    monitor="val_loss",
+                    patience=5,
+                    min_delta=0.05,
+                    mode='min',
+                )
+
+                pl_trainer_kwargs={"callbacks": [my_stopper]}
+            ..
+
+            Note that you can also use a custom PyTorch Lightning Trainer for training and prediction with optional
+            parameter ``trainer`` in :func:`fit()` and :func:`predict()`.
+        show_warnings
+            whether to show warnings raised from PyTorch Lightning. Useful to detect potential issues of
+            your forecasting use case. Default: ``False``.
+
+        References
+        ----------
+        .. [1] C. Challu et al. "N-HiTS: Neural Hierarchical Interpolation for Time Series Forecasting",
+               https://arxiv.org/abs/2201.12886
+        """
+        super().__init__(**self._extract_torch_model_params(**self.model_params))
+
+        # extract pytorch lightning module kwargs
+        self.pl_module_params = self._extract_pl_module_params(**self.model_params)
+
+        raise_if_not(
+            isinstance(layer_widths, int) or len(layer_widths) == num_stacks,
+            "Please pass an integer or a list of integers with length `num_stacks`"
+            "as value for the `layer_widths` argument.",
+            logger,
+        )
+
+        self.num_stacks = num_stacks
+        self.num_blocks = num_blocks
+        self.num_layers = num_layers
+        self.layer_widths = layer_widths
+
+        # Currently batch norm is not an option as it seems to perform badly
+        self.batch_norm = False
+
+        self.dropout = dropout
+
+        # Check pooling and downsampling numbers or compute good defaults
+        sizes = NHiTS._prepare_pooling_downsampling(
```

Comment:
Don't you think that as `_prepare_pooling_downsampling` is static we could (should?) use it as `NHiTS` method rather than `self` ?

Reply:
It's a minor thing but I would call it with `self` from within the class.

- with `self`, when renaming the class, we don't have to refactor the static method calls
- a more general thing: if we for example implement a subclass of NHiTS and want to overwrite `_prepare_pooling_downsampling`, then without `self`, it would call the `_prepare_pooling_downsampling` of the parent class, and not the new one.

But yeah for the current state there is no difference

Reply:
OK fair enough, those are good points.